adapt/src/main/java/io/github/vshnv/adapt/adapter/AdaptAdapter.kt

```diff
@@ -1,11 +1,15 @@
 package io.github.vshnv.adapt.adapter
 
+import android.widget.Filter
+import android.widget.Filterable
 import androidx.recyclerview.widget.RecyclerView
 
-abstract class AdaptAdapter<T>: RecyclerView.Adapter<AdaptViewHolder<T>>() {
+abstract class AdaptAdapter<T>: RecyclerView.Adapter<AdaptViewHolder<T>>(), Filterable {
     abstract val currentList: List<T>
+    abstract var searchFilterable: Filter?
```

Comment:
This shouldn't ideally be exposed. 

Ideally we should try to implement the `Filter` as a DSL into AdaptScope and hide the instance in the implementation.

Reply:
I think you can check my other branch "**i1**" in fork repo.

there i already added the filter in AdaptScope